analysis/dataset_definition_med_status_data_development.py

```diff
@@ -0,0 +1,124 @@
+time_interval = months(6)
+
+pharmacy_first_ids = clinical_events.where(
+    clinical_events.snomedct_code.is_in(pharmacy_first_event_codes)
+).consultation_id
+
+
+# Select medications
+# Pre launch, any medication
+selected_medications_any_pre = medications.where(
+    medications.date.is_on_or_between(
+        pharmacy_first_launch_date - time_interval,
+        pharmacy_first_launch_date - days(1),
+    )
+).sort_by(medications.date)
+
+# Pre launch, pharmacy first medication (pfmed)
+selected_medications_pfmed_pre = (
+    medications.where(
+        medications.date.is_on_or_between(
+            pharmacy_first_launch_date - time_interval,
+            pharmacy_first_launch_date - days(1),
+        )
+    )
+    .where(medications.dmd_code.is_in(pharmacy_first_med_codes))
+    .sort_by(medications.date)
+)
+
+# Pre launch, pharmacy first id (pdid) identified through clinical events
+# and pharmacy first medication (pfmed)
+selected_medications_pfmed_pfid_pre = (
+    medications.where(
+        medications.date.is_on_or_between(
+            pharmacy_first_launch_date - time_interval,
+            pharmacy_first_launch_date - days(1),
+        )
+    )
+    .where(medications.dmd_code.is_in(pharmacy_first_med_codes))
+    .where(medications.consultation_id.is_in(pharmacy_first_ids))
+    .sort_by(medications.date)
+)
+
+# Post launch, any medication
+selected_medications_any_post = medications.where(
+    medications.date.is_on_or_between(
+        pharmacy_first_launch_date, pharmacy_first_launch_date + time_interval
+    )
+).sort_by(medications.date)
+
+# Post launch, pharmacy first medication (pfmed)
+selected_medications_pfmed_post = (
+    medications.where(
+        medications.date.is_on_or_between(
+            pharmacy_first_launch_date, pharmacy_first_launch_date + time_interval
+        )
+    )
+    .where(medications.dmd_code.is_in(pharmacy_first_med_codes))
+    .sort_by(medications.date)
+)
+
+# Post launch, pharmacy first id (pdid) identified through clinical events
+# and pharmacy first medication (pfmed)
+selected_medications_pfmed_pfid_post = (
+    medications.where(
+        medications.date.is_on_or_between(
+            pharmacy_first_launch_date, pharmacy_first_launch_date + time_interval
+        )
+    )
+    .where(medications.dmd_code.is_in(pharmacy_first_med_codes))
+    .where(medications.consultation_id.is_in(pharmacy_first_ids))
+    .sort_by(medications.date)
+)
+
+selected_medications_dict = {
+    "pre_any": selected_medications_any_pre,
+    "pre_pfmed": selected_medications_pfmed_pre,
+    "pre_pfmedid": selected_medications_pfmed_pfid_pre,
+    "post_any": selected_medications_any_post,
+    "post_pfmed": selected_medications_pfmed_post,
+    "post_pfmedid": selected_medications_pfmed_pfid_post,
+}
+
+# Count all medication status
+# This will add 6 x 28 = 168 new columns
+for desc, selected_medications in selected_medications_dict.items():
+    for status in range(29):
```

Comment:
Just to note here for anyone else looking - this is as we expect medication status to be >=0 and <=28. https://docs.opensafely.org/ehrql/reference/schemas/raw.tpp/#medications.medication_status
We iterate to check that there aren't any values not listed in the documentation.